modules/experimental/src/gpgpu/point-in-polygon/gpu-point-in-polygon.js

```diff
@@ -0,0 +1,183 @@
+    }
+
+    const {vertices, indices, vertexCount, ids} = triangulatePolygons(polygons);
+    this._updateResources(vertices, indices, ids, vertexCount);
+  }
+
+  filter({positionBuffer, filterValueIndexBuffer, count}) {
+    this.filterTransform.update({
+      sourceBuffers: {
+        a_position: positionBuffer
+      },
+      feedbackBuffers: {
+        filterValueIndex: filterValueIndexBuffer
+      },
+      elementCount: count
+    });
+    const {polygonTexture, boundingBox} = this;
+
+    this.filterTransform.run({
+      moduleSettings: {boundingBox, texture: polygonTexture}
+    });
+  }
+
+  // PRIVATE
+
+  _setupResources() {
+    const {gl} = this;
+
+    // texture to render polygons to
+    this.polygonTexture = new Texture2D(gl, {
+      format: GL.RGB,
+      type: GL.UNSIGNED_BYTE,
+      dataFormat: GL.RGB,
+      border: 0,
+      mipmaps: false,
+      parameters: {
+        [GL.TEXTURE_MAG_FILTER]: GL.NEAREST,
+        [GL.TEXTURE_MIN_FILTER]: GL.NEAREST,
+        [GL.TEXTURE_WRAP_S]: gl.CLAMP_TO_EDGE,
+        [GL.TEXTURE_WRAP_T]: gl.CLAMP_TO_EDGE
+      }
+    });
+    this.positionBuffer = new Buffer(gl, {accessor: {type: GL.FLOAT, size: 2}});
+    this.idBuffer = new Buffer(gl, {accessor: {type: GL.FLOAT, size: 1}});
+    this.indexBuffer = new Buffer(gl, {
+      target: GL.ELEMENT_ARRAY_BUFFER,
+      accessor: {type: GL.UNSIGNED_SHORT}
+    });
+
+    // transform to generate polygon texture
+    this.polyTextureTransform = new Transform(gl, {
+      id: `polygon-texture-creation-transform`,
+      elementCount: 0,
+      _targetTexture: this.polygonTexture,
+      _targetTextureVarying: 'v_polygonColor',
+      vs: POLY_TEX_VS,
+      drawMode: GL.TRIANGLES,
+      isIndexed: true,
+      sourceBuffers: {
+        a_position: this.positionBuffer,
+        a_polygonID: this.idBuffer,
+        indices: this.indexBuffer
+      }
+    });
+
+    // transform to perform filtering
+    this.filterTransform = new Transform(gl, {
+      id: 'filter transform',
+      vs: FILTER_VS,
+      modules: [textureFilterModule],
+      varyings: ['filterValueIndex']
+    });
+  }
+
+  _updateResources(vertices, indices, ids, vertexCount) {
+    const boundingBox = getBoundingBox(vertices, vertexCount);
+    const [xMin, yMin, xMax, yMax] = boundingBox;
+    const width = xMax - xMin;
+    const height = yMax - yMin;
+    const whRatio = width / height;
+    const {textureSize} = this;
+
+    // calculate max texture size with same aspect ratio
+    let texWidth = textureSize;
+    let texHeight = textureSize;
+    if (whRatio > 1) {
+      texHeight = texWidth / whRatio;
+    } else {
+      texWidth = texHeight * whRatio;
+    }
+
+    this.boundingBox = boundingBox;
+    this.polygonTexture.resize({width: texWidth, height: texHeight, mipmaps: false});
+    this.positionBuffer.setData(new Float32Array(vertices));
+    this.idBuffer.setData(new Float32Array(ids));
+    this.indexBuffer.setData(new Uint16Array(indices));
```

Comment:
Any chance this polygon could get bigger than 65k vertices? Maybe worth putting an assert here?

Reply:
Added an assert where these indices are generated.